batch/add-application/add-application.sh

```diff
@@ -1,49 +1,80 @@
 #!/bin/bash
+# Failed validation in Cloud Shell on 4/7/2022
+
+# <FullScript>
+# Create a Batch account in Batch service mode
+
+# Variable block
+let "randomIdentifier=$RANDOM*$RANDOM"
+location="East US"
+resourceGroup="msdocs-batch-rg-$randomIdentifier"
+storageAccount="msdocsstorage$randomIdentifier"
+batchAccount="msdocsbatch$randomIdentifier"
 
 # Create a resource group.
-az group create --name myResourceGroup --location westeurope
+az group create --name $resourceGroup --location westeurope
 
 # Create a general-purpose storage account in your resource group.
 az storage account create \
-    --resource-group myResourceGroup \
-    --name mystorageaccount \
+    --resource-group $resourceGroup \
+    --name $storageAccount \
     --location eastus \
     --sku Standard_LRS
 
 # Create a Batch account.
 az batch account create \
-    --name mybatchaccount \
-    --storage-account mystorageaccount \
-    --resource-group myResourceGroup \
+    --name $batchAccount \
+    --storage-account $storageAccount \
+    --resource-group $resourceGroup \
     --location eastus
 
 # Authenticate against the account directly for further CLI interaction.
 az batch account login \
-    --name mybatchaccount \
-    --resource-group myResourceGroup \
+    --name $batchAccount \
+    --resource-group $resourceGroup \
     --shared-key-auth
 
 # Create a new application.
 az batch application create \
-    --resource-group myResourceGroup \
-    --name mybatchaccount \
-    --application-name "My Application"
+    --resource-group $resourceGroup \
+    --name $batchAccount \
+    --application-name "MyApplication" # ?? must exist?
+
+# Error message
+(InvalidUri) The requested URI does not represent any resource on the server.
```

Comment:
Application names cannot contain spaces, so removing the empty space should resolve the InvalidURI error.